app/scripts/lib/rpc-method-middleware/handlers/get-locale.js

```diff
@@ -0,0 +1,56 @@
+import { MESSAGE_TYPE } from '../../../../../shared/constants/app';
+
+const getLocaleOptions = {
+  methodNames: [MESSAGE_TYPE.GET_LOCALE],
+  implementation: getLocaleHandler,
+  hookNames: {
+    getWalletLocale: true,
+    sendMetrics: true,
+  },
+};
+export default getLocaleOptions;
+
+/**
+ * @typedef {Object} getLocaleOptions
+ * @property {Function} getWalletLocale - The wallet_getLocale implementation
+ * @property {Function} sendMetrics - A function that registers a metrics event.
+ */
+
+/**
+ * @typedef {Object} getLocaleParam
+ * @property {string} origin - The origin url
+ */
+
+/**
+ * @param {import('json-rpc-engine').JsonRpcRequest<WatchAssetParam>} req - The JSON-RPC request object.
+ * @param {import('json-rpc-engine').JsonRpcResponse<true>} res - The JSON-RPC response object.
+ * @param {Function} _next - The json-rpc-engine 'next' callback.
+ * @param {Function} end - The json-rpc-engine 'end' callback.
+ * @param {getLocaleOptions} options
+ */
+
+async function getLocaleHandler(
+  req,
+  res,
+  _next,
+  end,
+  { getWalletLocale, sendMetrics },
+) {
+  const { origin } = req;
+  try {
+    const walletLocale = await getWalletLocale();
+
+    sendMetrics({
+      event: 'Get locale',
+      category: 'Permission',
+      referrer: {
+        url: origin,
+      },
+    });
```

Comment:
[REQ] Unless there are specific props etc we want to track in RPC method handlers, the createRPCMethodTrackingMiddleware file sets up tracking for most RPC methods. If its important to know everytime this method is called (versus a sampling), we would add it to the USER_PROMPTED_EVENT_NAME_MAP in that file ... i don't believe it is necessary to capture every event in this case but I could be wrong.